cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/ExpressionHandler.kt

```diff
@@ -502,111 +360,22 @@ class ExpressionHandler(lang: JavaLanguageFrontend) :
     ): de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression? {
         val nameExpr = expr.asNameExpr()
 
-        // TODO this commented code breaks field accesses to fields that don't have a primitive
-        // type.
-        //  How should this be handled correctly?
-        //    try {
-        //      ResolvedType resolvedType = nameExpr.calculateResolvedType();
-        //      if (resolvedType.isReferenceType()) {
-        //        return newReference(
-        //            nameExpr.getNameAsString(),
-        //            new Type(((ReferenceTypeImpl) resolvedType).getQualifiedName()),
-        //            nameExpr.toString());
-        //      }
-        //    } catch (
-        //        UnsolvedSymbolException
-        //            e) { // this might throw, e.g. if the type is simply not defined (i.e., syntax
-        // error)
-        //      return newReference(
-        //          nameExpr.getNameAsString(), new Type(UNKNOWN_TYPE), nameExpr.toString());
-        //    }
-        val name = this.parseName(nameExpr.nameAsString)
-        return try {
+        // Try to resolve it. We will remove this in a future where we do not really in the
```

Comment:
Is the `TODO` of this method resolved? If so, please delete it.

Also, please document the method according to the new standards: https://github.com/Fraunhofer-AISEC/cpg/pull/1683#discussion_r1752563555

This sentence is broken